src/app/Models/Duplicate.php

```diff
@@ -158,4 +160,12 @@ public function person2() {
     public function commandRun() {
         return $this->belongsTo(CommandRun::class);
     }
+
+    /**
+     * Relation with MatchingAlgorithm
+     * @return BelongsTo
+     */
+    public function matchingAlgorithm(){
```

Comment:
Why do you need this ?

Reply:
Isn't it necessary to create a relationship between Duplicate and MatchingAlgorithm to be able to use the computeSimilarity function ?

Reply:
To better understand what are relationship, take a look at this : [https://laravel.com/docs/10.x/eloquent-relationships#introduction](https://laravel.com/docs/10.x/eloquent-relationships#introduction).

Relationship in your model are used to mimics relationship in your database